### PR TITLE
Make payment request expiry configurable

### PR DESCRIPTION
**Pull request by btcontract**

### CODE REVIEW

**Comment by btcontract** [comment]
Also as far as I understand unpaid payments don't currently get removed from `LocalPaymentHandler`'s map, this may be used as an attack vector for nodes who issue payment requests on demand over the internet. The latest commit fixes that.

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ case class PaymentRequest(prefix: String, amount: Option[MilliSatoshi], timestam
 
   def routingInfo(): Seq[Seq[ExtraHop]] = tags.collect { case t: RoutingInfoTag => t.path }
 
-  def expiry: Option[Long] = tags.collectFirst {
+  lazy val expiry: Option[Long] = tags.collectFirst {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala]: we should probably `lazy val` all the other `def`

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala [hunk 2]
```diff
@@ -5,25 +5,32 @@ import fr.acinq.bitcoin.{BinaryData, Crypto, MilliSatoshi}
...
-class LocalPaymentHandler(nodeParams: NodeParams) extends Actor with ActorLogging {
+class LocalPaymentHandler(nodeParams: NodeParams)(implicit ec: ExecutionContext = ExecutionContext.Implicits.global) extends Actor with ActorLogging {
+
+  context.system.scheduler.schedule(10 minutes, 10 minutes)(self ! Platform.currentTime / 1000)
 
   override def receive: Receive = run(Map())
 
   def run(h2r: Map[BinaryData, (BinaryData, PaymentRequest)]): Receive = {
 
+    case currentSeconds: Long =>
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala]: we should also add a maximum number of pending requests in the future if we really want to be safe

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala [hunk 3]
```diff
@@ -5,25 +5,32 @@ import fr.acinq.bitcoin.{BinaryData, Crypto, MilliSatoshi}
...
+      context.become(run(h2r.filter { case (_, (_, pr)) => pr.expiry.exists(_ + pr.timestamp > currentSeconds) }))
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala]: How about that instead? maybe more explicit?
```scala
case currentSeconds: Long =>
      context.become(run(h2r.collect {
        case e@(_, (_, pr)) if pr.expiry.isEmpty => e // requests that don't expire are kept forever
        case e@(_, (_, pr)) if pr.timestamp + pr.expiry.get > currentSeconds => e // clean up expired requests
      }))
```

**Comment by btcontract** [comment]
I already need this for my use case but I can just use a fork for now.

**Comment by pm47** [comment]
I have taken the liberty to push a few fixes on your branch, mainly:
- made `payment-request-expiry` a duration
- renamed `max-pending-requests` -> `max-pending-payment-requests`
- changed type of  `max-pending-payment-requests` `Long`->`Int`
- made payment handler return an error (e.g. to API) when we are over limit

**Comment by pm47** [comment]
@btcontract if that's OK with you I'm ready to merge

**Comment by btcontract** [comment]
Yes, it's totally fine, thanks for improving it!

**Pull request merged**